src/common.rs

```diff
@@ -71,14 +71,18 @@ pub fn get_local_components(
         let component_name: crate::socialdb_types::ComponentName = component_filepath
             .strip_prefix("src")?
             .with_extension("")
-            .to_str()
-            .wrap_err_with(|| {
-                format!(
-                    "Component name cannot be presented as UTF-8: {}",
-                    component_filepath.display()
-                )
-            })?
-            .replace('/', ".");
+            .components()
+            .filter_map(|component| match component {
+                std::path::Component::Normal(text) => Some(text.to_str().wrap_err_with(|| {
+                    format!(
+                        "Component name cannot be presented as UTF-8: {}",
+                        component_filepath.display()
+                    )
+                }).ok()?),
```

Comment:
There is no point in doing wrap_err and ? given that you convert it to Option with ok() anyway